### DISCUSSION TITLE
New open Source "Google Photo killer" immich.app

### DISCUSSION THREAD

**Discussion by elversoee** [tteck/Proxmox#1096] [Ideas]
Hope this relativity new App will be supported

https://immich.app/

https://github.com/immich-app/immich

already have Android and IOS apps. still under very active development.
but hope for another great and simple install script 

alternative version without Machine Learning
https://github.com/imagegenius/docker-immich

https://www.reddit.com/r/selfhosted/comments/10h5ur0/immich_is_too_good_what_is_the_catch_for_all_of_us/

**Comment by DocDrydenn** [comment]
+1

I'm currently stuck running this as a Snap since I'm not running anything Docker. Unfortunately, the snap usually lags behind the Immich releases... Getting this as a Proxmox LXC would be awesome.

**Reply by fantasjr** [reply]
+1

**Comment by ka2er** [comment]
+1

**Comment by KinDR007** [comment]
+1

**Comment by Slipperyclock** [comment]
+1

**Comment by emddummi** [comment]
+1

**Comment by r4ds3c** [comment]
+1

**Comment by DocDrydenn** [comment]
To get this added as a Proxmox LXC script, we need a Debian install path (i.e. "bare-metal" install). These "+1" are great, but the real pressure needs to be put on Immich [here](https://github.com/immich-app/immich/discussions/1657).

**Comment by johndoe0815** [comment]
I managed to set it up in a docker LXC from tteck as per official instructions from immich page. Beware however that it does NOT work with the docker alpine LXC, only with the "full" one. The alpine version throws an error when deploying the stack on the line
`- /etc/localtime:/etc/localtime:ro`

**Comment by Smile2Grow** [comment]
There are two other alternatives:

1. as tteck already wrote, install LXC dockge, during the installation you will be asked if you want to install Immich.
2. install lxc docker and then https://runtipi.io/ using the command curl -L https://setup.runtipi.io | bash. You can use runtipi to install various other interesting applications.

**Reply by mastan30** [reply]
Hi I was able to install using dockge, but I am having issue giving unpriviliged container read write access using lxc id mapping, Can someone help me with the id mapping, I know this is not the right forum but looking for some help.

**Comment by mastan30** [comment]
Finally able to get immich running in a docker unprivileged LXC, Here's the step by step guide to do it.


# Immich Docker Install

1. To install immich, the recommended way to install is using a docker compose, we can't install it on a bare metal debian os. So we will install a docker LXC using the proxmox script and install docker compose.Then we will create the docker-compose file for installing immich.

2. To install docker LXC we can used the following Proxmox script from the beloved tteck github:

		

		Docker LXC
		Options to Install Portainer and/or Docker Compose V2
		Docker is an open-source project for automating the deployment of applications as portable, self-sufficient containers.

		🛈 If the LXC is created Privileged, the script will automatically set up USB passthrough.

		To create a new Proxmox VE Docker LXC, run the command below in the Proxmox VE Shell.

		Copy
		bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/docker.sh)"
		⚡ Default Settings: 2GB RAM - 4GB Storage - 2vCPU ⚡
		As an alternative option, you can use Alpine Linux and the Docker package to create a Docker LXC container with faster creation time and minimal system resource usage.

		To create a new Proxmox VE Alpine-Docker LXC, run the command below in the Proxmox VE Shell.

		Copy
		bash -c "$(wget -qO - https://github.com/tteck/Proxmox/raw/main/ct/alpine-docker.sh)"
		⚡ Default Settings: 1GB RAM - 2GB Storage - 1vCPU ⚡
		⚠ Run Compose V2 by replacing the hyphen (-) with a space, using docker compose, instead of docker-compose.

		Portainer Interface: IP:9443


3. Once the container is installed, let's mount a mountpoint of SMB share where the images are uploaded.

4. In order to write the files into the SMB share, there's a catch here. The LXC we created using the script is an unpriviliged container which is the safest way to build container but the issue is we can't write anything into the shared drived from Proxmox host.

5. Instead of mounting the SMB share via Proxmox UI, we use fstab to mount the SMB share. Here are the steps to mount

	```
	  vi /etc/fstab
	```
	
	Then add the following line replacing the uid and gid for which you want the access to the proxmox container
	
	```
	  //192.168.X.XX/Path/to/share/ /path/to/mount/       cifs    noperm,iocharset=utf8,rw,credentials=/root/.storage_credentials,uid=1000,gid=1000,file_mode=0660,dir_mode=0770 0 0
	```
     eg: //192.168.1.19/Photos/immichtest/ /mnt/pve/immichtest/       cifs    noperm,iocharset=utf8,rw,credentials=/root/.storage_credentials,uid=1000,gid=1000,file_mode=0660,dir_mode=0770 0 0
	 
	```
      vi /root/.storage_credentials
    ```	
	 In this file add the username and password as below:
	 
	```
	username=username
	password=password
	```
	
	```
	mount -a
	```
   This will mount the SMB share into the Proxmox host, now check the mount directory permissions by doing ls -al and check if the mount has the permissions of assigned uid and gid. If not do a reboot of the host and it takes care of it.
   
6. Shutdown the container and mount the SMB share to the proxmox LXC by following commands:

   ```
    pct set 100 -mp0 /mnt/bindmounts/shared,mp=/shared
   
   ```
    eg: pct set 100 -mp0 /mnt/pve/immichtest/ ,mp=/immichtest/   - 100 is the lxc id
   
7. Start the container and check the if the mount point is assigned with right uid and gid inside the container.

8. Now in the LXC create a folder using following commands

   ```
     mkdir ./immich/
	 cd ./immich/
   ```
   
   create the docker-compose file and also the .env file in the directory as per the immich guide
   
   ```
      https://immich.app/docs/install/docker-compose
   ```
   
   Run the docker compose and follow the immichi guide to setup your albums and libraries as required.
   
   
9. Optinally in order to add hardware acceleration, I used NVIDIA acceleration. In order to get it working here are the steps to follow:
   
   Next step is to passthrough the nvidia graphics card to the lxc, In order to do the passthrough follow the below guide.
   
   > [Nvidia graphics passthrough](https://gist.github.com/egg82/90164a31db6b71d36fa4f4056bbee2eb_
   > [Nivdia graphics passthrough video](https://www.youtube.com/watch?v=-Us8KPOhOCY)
   
   I have already installed NVIDIA drivers for jellyfin lxc in proxmox host, so I just installed the nvidia drivers in container using below commands:
   
   ```
    wget https://us.download.nvidia.com/XFree86/Linux-x86_64/535.146.02/NVIDIA-Linux-x86_64-{Latest-version}.run
   ```
    you can get the NVIDIA latest driver url from here : https://www.nvidia.com/Download/index.aspx
	
   once downloaded install the drivers (The --no-kernel-module module is used as already there's kernel installed ):
   ```
   chmod +x NVIDIA-Linux-x86_64-{Latest-version}.run
   
   ./NVIDIA-Linux-x86_64-{Latest-version}.run ----no-kernel-module
   ```
   
   once installed make sure the drivers are installed by running:
   ```
   nvidia-smi -l 10
   ```
   
10. Now we need to pass through the hardware acceleration to docker container by following the immich guide:
    
	```
	https://immich.app/docs/features/hardware-transcoding
    ```
	
	In order to have the driver passed to container, we need to have nvidia container toolkit installed using the following commands in the container:
	(https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html)
	
	```
	curl -fsSL https://nvidia.github.io/libnvidia-container/gpgkey | sudo gpg --dearmor -o /usr/share/keyrings/nvidia-container-toolkit-keyring.gpg \
    && curl -s -L https://nvidia.github.io/libnvidia-container/stable/deb/nvidia-container-toolkit.list | \
    sed 's#deb https://#deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://#g' | \
    sudo tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
	
	
	
	sed -i -e '/experimental/ s/^#//g' /etc/apt/sources.list.d/nvidia-container-toolkit.list
	
	
	
	sudo apt-get update
	
	
	sudo apt-get install -y nvidia-container-toolkit
	
	
	```
	
	Also one other step I needed to do to make sure the docker compose works with hardware-transcoding is to modify nvidia config using below commands in container:
	
	Edit /etc/nvidia-container-runtime/config.toml and disable cgroups:
	```
      
	  vi /etc/nvidia-container-runtime/config.toml
	  
	  set no-cgroups = true in the config.toml file, save and reboot the container. Now run the docker compose for immich.

	```
	
	
	
	References:
	
	https://immich.app/docs/install/script
	
	https://forum.proxmox.com/threads/how-to-automatically-mount-a-smb-share-on-an-lxc.118220/

    https://immich.app/docs/features/hardware-transcoding
	
	https://www.reddit.com/r/Proxmox/comments/s0ud5y/comment/jl4lef2/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button
	
	<blockquote class="reddit-embed-bq" data-embed-height="348"><a href="https://www.reddit.com/r/Proxmox/comments/s0ud5y/comment/jl4lef2/">Comment</a><br> by<a href="https://www.reddit.com/user/Masterofchicken/">u/Masterofchicken</a> from discussion<a href="https://www.reddit.com/r/Proxmox/comments/s0ud5y/cgroups2_problem_with_nvidiacontainercli/"><no value=""></no></a><br> in<a href="https://www.reddit.com/r/Proxmox/">Proxmox</a></blockquote><script async="" src="https://embed.reddit.com/widgets.js" charset="UTF-8"></script>

**Reply by fedeiglesias** [reply]
Thanks for the tutorial @mastan30, I wanted to know if it would make sense to activate the hardware acceleration if I only have a mobile processor, specifically an i7 12650H.

**Comment by keecy** [comment]
I would also have the immich has LXC container!
i will try from here, 
https://github.com/arter97/immich-native


Br,

**Comment by jonafriendj** [comment]
I have it running in docker and it works great. I'd love an LXC too. If I get some free time before it happens this would be my first attempt to create one.

**Comment by loeeeee** [comment]
I started with https://github.com/arter97/immich-native, and made myself a script, https://github.com/loeeeee/immich-in-lxc, that installs Immich without docker while also using CUDA. :)
